### PR TITLE
`num256` removal

### PR DESCRIPTION
**Pull request by AaronKutch**

### CODE REVIEW

**Comment by AaronKutch** [comment]
I was unfortunately unable to remove `num` because of a rational square root, I won't have time to replace it in the near future.

File: Cargo.toml
Comment (dzmitryhil) [inline Cargo.toml]: Don't forget to replace all links to your repo once we review and approve all the PRs.

File: Cargo.toml
Comment (dzmitryhil) [inline Cargo.toml]: Don't we need to update the "tokio" version ?

File: src/erc20_utils.rs
Comment (dzmitryhil) [inline src/erc20_utils.rs:43]: Does the logic here remain the same?

File: src/client.rs
Comment (leviathanbeak) [inline src/client.rs]: same as in clarity, a lot of unwraps here, would this make for a potential attack surface for our orchestrator?

File: src/client.rs
Comment (leviathanbeak) [inline src/client.rs]: why the braces after IF keyword?

File: Cargo.toml
Comment (AaronKutch) [reply Cargo.toml]: good catch

File: src/erc20_utils.rs
Comment (AaronKutch) [reply src/erc20_utils.rs:43]: `/ 2` may or may not be slow depending on how smart the compiler is. Because of the number of division and multiplication by two cases, I introduced `shr1` (shift right 1) and `shl1` (shift left 1) functions that do the same numerical thing more performantly.

File: src/client.rs
Comment (AaronKutch) [reply src/client.rs]: There is _less_ attack surface. Unchecked arithmetic wraps in release mode, and if an attacker was able to manipulate one of the two inputs of a multiplication to be large enough, they could get the output to be almost any number they wished. I'm not familiar enough with the code to know the implications, but I would rather that the relayer panic and halt than something worse happening.

File: src/client.rs
Comment (AaronKutch) [reply src/client.rs]: I personally always use explicit parenthesis. Sometimes I forget what order of operations Rust uses, or read an especially long if statement like this one the wrong way, so I think it is better to use explicit parenthesis.

File: Cargo.toml
Comment (AaronKutch) [reply Cargo.toml]: I think we have to do it one at a time, because using `/onomyprotocol/` doesn't work until the dependency PR is accepted

Comment (AaronKutch) [reply Cargo.toml]: I did allow maintainer changes, so if you want to merge more than one a day (we are 12 hours out of timezone sync) that's all you have to change

File: src/client.rs
Comment (leviathanbeak) [reply src/client.rs]: well that's the thing, we could/should then return an error so that (if needed) we exit out of it cleanly (or try to recover in some way) instead of panicking and breaking everything

File: src/client.rs
Comment (leviathanbeak) [reply src/client.rs]: I think we should follow the standard guidelines for the language, not personal preference

Comment (AaronKutch) [reply src/client.rs]: if you insist

File: src/client.rs
Comment (AaronKutch) [reply src/client.rs]: I have decided to return an error for that branch of the code, since the total supply in the world is much less than what a u128 can handle.

**Comment by AaronKutch** [comment]
I should detail my unwrapping strategy:
- If in a `#[test]` or function that will always be locally controlled, `unwrap`s can be used
- If there is something like a nonce or block number and we are adding or multiplying small constants to it, 2^256 is a very large number and `unwrap`s are warranted
- If the context means that values cannot overflow or indexes cannot be violated (e.g. if `from_bytes` is called on a slice with a constant range of `32..64`, it means `from_bytes` cannot fail)
- Something like ludicrous gas prices that shouldn't go beyond a certain level because of the finite amount of tokens in the system

If you see an unwrap that doesn't look right point it out

File: src/amm.rs
Comment (leviathanbeak) [inline src/amm.rs:121]: why is this getting unwrapped here?

**Comment by AaronKutch** [comment]
I made `seal_fields` into a `String` because it did not fit in a `Uint256` in general.

**Comment by AaronKutch** [comment]
There are also places where I replaced `f32` with `f64` because there's no reason not to use the higher precision

File: src/amm.rs
Comment (AaronKutch) [reply src/amm.rs:121]: `from_bytes_be` can fail if there are more than 32 bytes (and `num256` would just panic internally instead of giving a heads up). There is a checked `get(0..32)` range in the previous lines so it is ok

**Review by leviathanbeak — APPROVED** [review]

**Pull request merged**
